airflow/bin/cli.py

```diff
@@ -354,7 +374,8 @@ def run(args, dag=None):
         # via the UI (or vice versa) results in a permission error as the task
         # tries to write to a log file created by the other user.
         log_base = os.path.expanduser(conf.get('core', 'BASE_LOG_FOLDER'))
-        directory = log_base + "/{args.dag_id}/{args.task_id}".format(args=args)
+        log_relative_dir = logging_utils.get_log_directory(**locals())
```

Comment:
@aoen This is the part I don't like and it seems to me it is building additional legacy into airflow. But maybe I am not seeing it right. Im my opinion the task when it gets run just does a "import logging" and sets is name for logging. Based on that name logging will be configured. In this case here arbitrary files are created again and I am not seeing any improvement in handling this. One should never "get" the log directory, cause in many occasions it just might not make sense. How do I override the handler for logging for the tasks now? I do I configure it *not* to use a file on the filesystem?

In addition the use of "**locals" makes it very unclear what is used to setup the logging.

I understand that for this PR the change might be too broad to fix logging, but the additional legacy being added by `logging_utils.get_log_directory` I don't like.